app/src/main/java/org/mozilla/rocket/content/ContentRepository.kt

```diff
@@ -32,14 +31,19 @@ class ContentRepository {
             INSTANCE = null
         }
 
+        @JvmStatic
+        fun resetSubscriptionUrl(subscriptionUrl: String) {
+            INSTANCE?.setSubscriptionUrl(subscriptionUrl)
+        }
+
         @JvmStatic
         fun isEmpty() = INSTANCE == null
 
         private fun buildRepository(context: Context): Repository<out NewsItem> {
             return if (NewsSourceManager.getInstance().newsSource == NEWS_DB) {
-                RepositoryBhaskar(context)
+                RepositoryBhaskar(context, NewsSourceManager.getInstance().newsSourceUrl)
```

Comment:
Can we derive the URL from the current source and it's URL in AppConstant? I'll like to remove the NewsSourceManager in the future cause it lacks testing and make the URL/Source hard to sync.

Reply:
Maybe we could refine the class responsibility when we try to remove NewsSourceManager. Currently, it should be reasonable to keep url in NewsSourceManager.